### PR TITLE
[terra-functional-testing] Create a setLocale utility to update a terra-application's locale

### LINKED ISSUE #523: [terra-functional-testing] Create setLocale utility to override terra-application locale
# Feature Request
Create a Terra utility that sends a custom event to override the terra-application locale; this links up with terra-application's event listener api for testing [[1](https://github.com/cerner/terra-application/blob/main/src/application-base/private/useTestOverrides.js)]. 

This is necessary for full-stack test runs on deployed apps, and gives people the opportunity to update locales within a test without using the describe helper.

## Description

```js
Terra.setLocale('ar')
```

## Additional Context / Screenshots

https://github.com/cerner/terra-toolkit/pull/519#discussion_r554202926

### PR DESCRIPTION
**Pull request by yuderekyu**
### Summary

* Expose a utility that will update a terra-application's locale. This utility is attached to the Terra global and utilizes an event dispatch api under the hood.
* Use cases are updating locale on a case by case basis, and updating locale for deployed terra apps.
* Closes #523 
### Additional Details

Previous pr: https://github.com/cerner/terra-toolkit/issues/516
@cerner/terra

[CONTRIBUTORS.md]: ../blob/main/CONTRIBUTORS.md
[License]: ../blob/main/LICENSE

### CODE REVIEW

File: packages/terra-functional-testing/src/commands/utils/setLocale.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const dispatchCustomEvent = require('./dispatchCustomEvent');
+
+const logger = new Logger({ prefix: '[terra-functional-testing:setLocale]' });
+
+/**
+ * Updates terra-application's locale via custom event
+ * @param {string} locale - name of locale to update to
+ */
+const setLocale = (locale) => {
+  if (typeof locale === 'undefined' || !locale) {
```
Comment (StephenEsser) [inline packages/terra-functional-testing/src/commands/utils/setLocale.js]: The second check would probably cover both cases here, but nothing wrong with being extra precautions. 

image

**Review by StephenEsser — APPROVED** [review]

File: packages/terra-functional-testing/src/services/wdio-terra-service.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const {
   describeViewports,
   getViewports,
   hideInputCaret,
+  setLocale,
```
Comment (mjhenkes) [inline packages/terra-functional-testing/src/services/wdio-terra-service.js]: Should we be more verbose? setApplicationLocale?

File: packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx [hunk 3]
```diff
@@ -159,3 +159,14 @@ it('should hide the caret', () => {
...
 ```
+
+## Locale Update (Case by Case)
+
+### Terra.setLocale
+Use `Terra.setLocale` to update a terra application's locale on a case by case basis; particularly useful for testing locale changes on a deployed application. This method accepts a string containing the `locale` to update to.
+
+```js
+it('should update an application's locale, () => {
+  Terra.setLocale('en');
```
Comment (mjhenkes) [inline packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/wdio-testing/About.tool.mdx]: I like this syntax a lot. imo we should expose any other terra application switches the same way if a need arises (theme comes to mind). I don't we will find a reason to expose dispatch custom event itself at all.

We could use setLocale in the terra-dev-site project to eliminate a run.

**Review by benbcai — APPROVED** [review]

File: packages/terra-functional-testing/src/services/wdio-terra-service.js (hunk 2 above)
Comment (yuderekyu) [reply packages/terra-functional-testing/src/services/wdio-terra-service.js]: i can roll with that

Comment (yuderekyu) [reply packages/terra-functional-testing/src/services/wdio-terra-service.js]: renamed https://github.com/cerner/terra-toolkit/pull/524/commits/49b4d2cc78c3ef39d0f7d1d335c1b7a55c336f5d

**Review by mjhenkes — APPROVED** [review]

**Comment by yuderekyu** [comment]
looks like a new, "private" prop was merged into react-intl, causing snapshots to fail [[1](https://github.com/formatjs/formatjs/commit/82471d1aaeeecbb96cf6186c9ae66d25a884473d#)]. i assume we want to regenerate snapshots to bring the new prop in? pinging our resident intl expert @jmsv6d :)

**Pull request merged**